extractor/src/main/java/org/schabi/newpipe/extractor/stream/DeliveryFormat.java

```diff
@@ -0,0 +1,134 @@
+package org.schabi.newpipe.extractor.stream;
+
+import java.io.Serializable;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A class that is used to represent the way that a streaming service deliver their streams.
+ */
+public abstract class DeliveryFormat implements Serializable {
```

Comment:
Can you make this class an `interface` and add a `getUrl` method? This will save us some casts.